### PR TITLE
Implemented conformance tests for Enums chapter.

### PR DESCRIPTION
**Pull request by erictraut**

### CODE REVIEW

**Comment by erictraut** [comment]
Most of these tests are direct copies of the sample code in the spec. The only part that was a little tricky is that many of the type checker behaviors in the Enums chapter are optional, so many of the error conditions needed to be marked as `# E?`, and manual inspection was required to compile the "notes" section.

**Comment by erictraut** [comment]
@srittau, there appears to be a general problem with the pre-commit bot. The "CLA signing" rule thinks that it is a human who hasn't signed the CLA. Any thoughts on how we can fix this?

**Comment by JelleZijlstra** [comment]
I guess that's a bug in the bot, cc @ambv. It should ignore the pre-commit bot's email address somehow. We probably haven't run into this before because the main other repo where we use the bot is CPython, which doesn't use pre-commit autofixers.

I'm an admin on this repo and I can merge through the CLA bot failure, so I'll merge this (after I've reviewed it, which might take some time).

File: conformance/tests/enums_expansion.py
Comment (JelleZijlstra) [inline conformance/tests/enums_expansion.py:39]:
```suggestion
# > these enums allow flags to be combined in arbitrary ways.
```
Since the opening parentheses was omitted

File: conformance/tests/enums_expansion.py
Comment (JelleZijlstra) [inline conformance/tests/enums_expansion.py:49]:
```suggestion
    if f is CustomFlags.FLAG1 or f is CustomFlags.FLAG2:
```
I feel this makes the test a bit more interesting, since now a type checker that implements Flags naively should actually fail the test.

File: conformance/tests/enums_members.py
Comment (JelleZijlstra) [inline conformance/tests/enums_members.py:126]: I assume this is meant to be `Example2`, not `Example`.
```suggestion
Example2.__B  # E
```

File: conformance/tests/enums_expansion.py
Comment (JelleZijlstra) [reply conformance/tests/enums_expansion.py:49]: Oh, but apparently mypy does perform type narrowing on the code as you wrote it? That's bad.

**Comment by erictraut** [comment]
@JelleZijlstra, thanks for the review. I incorporated changes based on your suggestions.

I'm not able to override the "CLA Signing" rule, so I can't merge this. If/when you feel it's ready, please merge.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**
